examples/get-insights-data/main.py

```diff
@@ -0,0 +1,119 @@
+import os
+import sys
+import csv
+from utils.utils import PublicAPIClient, get_commandline_arguments
+
+
+def main():
+    args = get_commandline_arguments(
+        [
+            (
+                ["course_id"], {"type": int, "help": "id of the course to get insights for"},
+            )
+        ]
+    )
+    public_api_client = PublicAPIClient(args.subdomain)
+    perspectives = fetch_course_perspectives(public_api_client, args.course_id)
+    course_data = fetch_course_data(public_api_client, args.course_id)
+
+    construct_summary(public_api_client, perspectives, course_data)
+    construct_student_insights(public_api_client, perspectives, course_data)
+
+
+def fetch_course_perspectives(public_api_client, course_id):
+    response = public_api_client.request("get", f"/courses/{course_id}/perspectives")
+    return response.json()["perspectives"]
+
+
+def fetch_course_data(public_api_client, course_id):
+    response = public_api_client.request("get", f"/courses/{course_id}")
+    return response.json()["course"]
+
+
+def construct_summary(public_api_client, perspectives, course):
+    headers = [
+        "Course ID",
+        "Course Title",
+        "Perspective UUID",
+        "Perspective Title",
+        "Views",
+        "Time Viewed [min]",
+        "Unique Viewers",
+    ]
+    csv_data = []
+    for perspective in perspectives:
+        perspective_data = [
+            course["id"],
+            course["name"],
+            perspective["uuid"],
+            perspective["title"],
+        ]
+        summary_csv = public_api_client.request(
```

Comment:
I would rather have a helper function that parses CSV from the response, so we use the same here and in lines 96-99. Something like this:
```
def get_csv(public_api_client, url, parsed=False):
    response = public_api_client.request('get', url)
    content = response.content.decode('utf-8').splitlines()
    if parsed:
        return csv.DictReader(content)
    return csv.reader(content)
```

And then here we can have:
```
summary_csv = get_csv(public_api_client, f"perspectives/{perspective['uuid']}/insights/overview")
for row in summary_csv:
    if "Views" in row:
...
```

Note: use `row` instead of `i`

Reply:
Note: the whole iter_lines thingie comes from a Stackoverflow answer where they care about performance. But we don't have dozens of megabytes of CSVs, so I don't think that matters, and I find this more readable (also comes from SO). Not to mention: the original implementation converts the results into a list, which means the whole thing goes to memory, so performance goes out the window anyway. Again: a few MB of CSV can fit the memory of any conceivable computer nowadays. 😄

Reply:
done